pandas/core/indexes/datetimes.py

```diff
@@ -2704,7 +2715,49 @@ def date_range(start=None, end=None, periods=None, freq='D', tz=None,
     >>> pd.date_range(start='2017-01-01', end='2017-01-04', closed='right')
     DatetimeIndex(['2017-01-02', '2017-01-03', '2017-01-04'],
                   dtype='datetime64[ns]', freq='D')
+
+    Declare extra parameters (kwargs) to be used with the pd.to_datetime
+    function that is used when all three parameters `start`, `end`, and
+    `periods` are declared. If this results in anything else than a
+    DatetimeIndex (like in this example), you cannot specify `tz` or `name`.
+
+    >>> date_range('2018-04-24', '2018-04-27', periods=3, box=False)
+    array(['2018-04-24T00:00:00.000000000', '2018-04-25T12:00:00.000000000',
+           '2018-04-27T00:00:00.000000000'], dtype='datetime64[ns]')
     """
+
+    # See https://github.com/pandas-dev/pandas/issues/20808
```

Comment:
this needs to go in the DTI constructor itself. we already do some of this validation, it needs to fit in there. Further you don't need to worry about lots of other things that you are repeating, e.g. tz, which are already handled.